### PR TITLE
Support For Managing Reserved IP Addresses

### PR DESCRIPTION
**Pull request by AniJ98**
### **📝 Description**
This PR introduces comprehensive functionality and tests for Reserved IP Addresses in the Linodego client. The changes are necessary to provide robust support for managing Reserved IP addresses through the Linode API. The changes include:

Implementation of core Reserved IP operations:

- Listing Reserved IPs
- Getting a specific Reserved IP
- Reserving a new IP
- Deleting a Reserved IP

Test coverage:

- TestReservedIPAddresses_InsufficientPermissions: Verifies behavior when the user lacks necessary permissions.
- TestReservedIPAddresses_EndToEndTest: Performs a complete workflow test for users with proper permissions.
- TestReservedIPAddresses_ListIPAddressesVariants: Tests various scenarios for listing IP addresses.
- TestReservedIPAddresses_GetIPAddressVariants: Covers different cases for retrieving a specific IP address.
- TestReservedIPAddresses_ReserveIPVariants: Tests multiple scenarios for reserving an IP, including edge cases.
- TestReservedIPAddresses_DeleteIPAddressVariants: Verifies deletion functionality under various conditions.

### **✔️ How to Test**
What are the steps to reproduce the issue or verify the changes?

To verify the changes related to Reserved IP functionality, follow these steps:

1. Ensure you have a valid Linode API token with the "can_reserve_ip" permission enabled. Set up your environment: 

`export LINODE_TOKEN="your_token_here" `

Additionally you need to set the following if you want to test it in a different environment: 

```
export LINODE_URL="https://api.linode.com"
export LINODE_API_VERSION="v4beta"
```

Navigate to the test directory within the linodego project. Update the LINODE_TOKEN in the Makefile: 

`LINODE_TOKEN="your_token_here"`

2. Run the tests using one of the following commands: 

To run all integration tests: 

`make testint`

To run only Reserved IP related tests: 

`go test -v ./integration -run TestReservedIPAddresses `

To run a specific test:

`go test -v ./integration -run TestReservedIPAddresses_EndToEndTest`

3. Verify the test output for any failures or unexpected behavior.

Note:

Ensure you have proper permissions and sufficient quota in your Linode account to perform Reserved IP operations. Some tests may create and delete resources, so use a testing environment if possible.

### CODE REVIEW

File: test/integration/network_reserved_ips_test.go [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+		}
+		reservedIPs = append(reservedIPs, reserveIP.Address)
+		t.Logf("Successfully reserved IP %d: %s", i+1, reserveIP.Address)
+	}
+
+	// Verify that we can't reserve more IPs
+	_, exceedErr := client.ReserveIPAddress(context.Background(), ReserveIPOptions{Region: "us-east"})
+	if exceedErr == nil {
+		t.Errorf("Expected error when exceeding reservation limit, got nil")
+	}
```
Comment (yec-akamai) [inline test/integration/network_reserved_ips_test.go:302]: The reservation limit seems can be expanded. I think we might want to remove this verification to avoid unexpected failure.

**Review by yec-akamai — COMMENTED** [review]
LGTM. Great test coverage! Just one small comment:

File: test/integration/network_reserved_ips_test.go (hunk 1 above)
Comment (AniJ98) [reply test/integration/network_reserved_ips_test.go]: Removed this check and re-recorded the fixture for this test.

Comment (ykim-1) [reply test/integration/network_reserved_ips_test.go]: I think it is still a valid test coverage. I pushed a commit separating into a new test case.

**Review by yec-akamai — APPROVED** [review]
Tested and works well on my end, thank you for the contribution!

File: test/integration/network_reserved_ips_test.go (hunk 1 above)
Comment (yec-akamai) [reply test/integration/network_reserved_ips_test.go]: The new test case works well, thank you @ykim-1!

**Review by ykim-1 — APPROVED** [review]
LGTM, test pass locally and thank you for the contribution!

**Review by ezilber-akamai — APPROVED** [review]
Nice work! Tests are passing on my end. Thanks for the contribution!

**Review by yec-akamai — APPROVED** [review]
